src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php

```diff
@@ -131,12 +131,21 @@ public function handle(AddProductToOrderCommand $command)
 
         // update totals amount of order
         // @todo: use https://github.com/PrestaShop/decimal for prices computations
-        $order->total_products += (float) $cart->getOrderTotal(false, Cart::ONLY_PRODUCTS);
-        $order->total_products_wt += (float) $cart->getOrderTotal(true, Cart::ONLY_PRODUCTS);
+        $order->total_products = (float) $cart->getOrderTotal(false, Cart::ONLY_PRODUCTS);
+        $order->total_products_wt = (float) $cart->getOrderTotal(true, Cart::ONLY_PRODUCTS);
 
-        $order->total_paid += Tools::ps_round((float) $cart->getOrderTotal(true, $totalMethod), 2);
-        $order->total_paid_tax_excl += Tools::ps_round((float) $cart->getOrderTotal(false, $totalMethod), 2);
-        $order->total_paid_tax_incl += Tools::ps_round((float) $cart->getOrderTotal(true, $totalMethod), 2);
+        $order->total_paid = Tools::ps_round(
+            (float) $cart->getOrderTotal(true, $totalMethod),
+            $this->context->getComputingPrecision()
```

Comment:
I don't think you should use the precision rom context It will use the context's currency which might be different from the order one Rather use a dedicated ComputingPrecision object along with the precision o the currency in the order

Reply:
Thanks. I applied your review ;) (Thanks a lot dude ;))